src/Ldap/LdapData.php

```diff
@@ -0,0 +1,146 @@
+        $affiliationCollection = collect(($data['cornelleduaffiliation'] ?? null) ?: null);
+        $affiliations = $affiliationCollection->toArray();
+
+        // Use the defined primary affiliation or the first affiliation in the collection.
+        $primaryAffiliation = ($data['cornelleduprimaryaffiliation'] ?? null) ?: $affiliationCollection->shift() ?? '';
+
+        // Secondary affiliation is the first affiliation that is not the primary affiliation.
+        $secondaryAffiliation = $affiliationCollection->reject($primaryAffiliation)->first() ?? '';
+
+        // Process previousNetids: always convert the comma-separated string to an array.
+        $previousNetIds = collect(explode(',', $data['cornelledupreviousnetids'] ?? ''))
+            ->map(fn ($id) => trim($id))->filter()->all();
+
+        // Same for previousEmplids
+        $previousEmplids = collect(explode(',', $data['cornelledupreviousemplids'] ?? ''))
+            ->map(fn ($id) => trim($id))->filter()->all();
+
+        // User may have exercised FERPA right to suppress name.
+        if (empty($firstName) && empty($lastName)) {
+            $firstName = 'Cornell';
+            $lastName = $primaryAffiliation === 'student' ? 'Student' : 'User';
+        }
+
+        // Format an array to match the old LDAP::data function
+        $ldapData = [
+            'emplid' => $data['cornelleduemplid'] ?? '',
+            'firstname' => $firstName ?? '',
+            'lastname' => $lastName ?? '',
+            'name' => trim("$firstName $lastName"),
+            'email' => $data['mail'] ?? '',
+            'campusphone' => $data['cornelleducampusphone'] ?? '',
+            'netid' => $data['uid'],
+            'deptname' => $data['cornelledudeptname1'] ?? '',
+            'primaryaffiliation' => $primaryAffiliation,
+            'secondaryaffiliation' => $secondaryAffiliation,
+            'wrkngtitle' => $data['cornelleduwrkngtitle1'] ?? '',
+            'affiliations' => $affiliations,
+            'previousnetids' => $data['cornelledupreviousnetids'] ?? null,
+            'previousemplids' => $data['cornelledupreviousemplids'] ?? null,
+        ];
+
+        return new LdapData(
+            uid: $data['uid'],
+            principalName: $data['edupersonprincipalname']
+                ?? $data['uid'].'@cornell.edu',
+            emplid: $data['cornelleduemplid'] ?? '',
+
+            firstName: $firstName,
+            lastName: $lastName,
+
+            // Use preferred display name if it is not null, otherwise fall back on first_name + last_name.
+            displayName: $data['displayname']
+                ?? trim($firstName.' '.$lastName),
+
+            // Only set 'email' if it is not empty.
+            email: ($data['mail'] ?? null) ?: null,
+
+            campusPhone: $data['cornelleducampusphone'] ?? null,
+            deptName: $data['cornelledudeptname1'] ?? null,
+            workingTitle: $data['cornelleduwrkngtitle1'] ?? null,
```

Comment:
if an application needs department name and working title, then these data should be coming from Workday not from LDAP

Reply:
I think you are bringing up a good question about the source of the data. I can see that the [attribute documentation](https://confluence.cornell.edu/spaces/IDM/pages/104400024/Attributes) indicates the those fields come from "HR" (i.e., Workday), but I don't know much else.

Do we have reason to be concerned about some of these pass-through fields? Looking at the sources for the attributes, lots of them come from Workday and Peoplesoft.